crates/frontend/src/pages/dimensions.rs

```diff
@@ -64,13 +65,26 @@ pub fn dimensions() -> impl IntoView {
                 open_drawer("dimension_drawer");
             };
 
-            let edit_icon: HtmlElement<html::I> =
-                view! { <i class="ri-pencil-line ri-xl text-blue-500"></i> };
+            let handle_delete = move |_| {
+                let tenant = tenant_rs.get();
+                let name = dimesnion_name.get_value();
+                spawn_local({
+                    async move {
+                        let _ = delete_dimension(name, tenant).await;
+                        dimensions_resource.refetch();
+                    }
+                });
+            };
 
             view! {
-                <span class="cursor-pointer" on:click=edit_click_handler>
-                    {edit_icon}
-                </span>
+                <div class="join">
+                    <span class="cursor-pointer" on:click=edit_click_handler.clone()>
```

Comment:
why this extra cloning of the function?
this was not there earlier

Reply:
removed , it got added by mistake